lib/ansible/modules/cloud/amazon/route53.py

```diff
@@ -377,135 +382,134 @@ def commit(changes, retry_interval, wait, wait_timeout):
+    identifier_in = module.params.get('identifier')
+    weight_in = module.params.get('weight')
+    region_in = module.params.get('region')
+    health_check_in = module.params.get('health_check')
+    failover_in = module.params.get('failover')
+    vpc_id_in = module.params.get('vpc_id')
+    wait_in = module.params.get('wait')
+    wait_timeout_in = module.params.get('wait_timeout')
 
     region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module)
 
-    value_list = ()
-
-    if isinstance(value_in, str):
-        if value_in:
-            value_list = sorted([s.strip() for s in value_in.split(',')])
-    elif isinstance(value_in, list):
-        value_list = sorted(value_in)
-
     if zone_in[-1:] != '.':
         zone_in += "."
 
     if record_in[-1:] != '.':
         record_in += "."
 
     if command_in == 'create' or command_in == 'delete':
-        if not value_in:
-            module.fail_json(msg = "parameter 'value' required for create/delete")
-        elif alias_in:
-            if len(value_list) != 1:
-                module.fail_json(msg = "parameter 'value' must contain a single dns name for alias create/delete")
-            elif not alias_hosted_zone_id_in:
-                module.fail_json(msg = "parameter 'alias_hosted_zone_id' required for alias create/delete")
-        elif ( weight_in is not None or region_in is not None or failover_in is not None ) and identifier_in is None:
-            module.fail_json(msg= "If you specify failover, region or weight you must also specify identifier")
-
-    if command_in == 'create':
-        if ( weight_in is not None or region_in is not None or failover_in is not None ) and identifier_in is None:
-            module.fail_json(msg= "If you specify failover, region or weight you must also specify identifier")
-        elif ( weight_in is None and region_in is None and failover_in is None ) and identifier_in is not None:
-            module.fail_json(msg= "You have specified identifier which makes sense only if you specify one of: weight, region or failover.")
-
-
+        if alias_in and len(value_in) != 1:
+            module.fail_json(msg="parameter 'value' must contain a single dns name for alias records")
+        if (weight_in is not None or region_in is not None or failover_in is not None) and identifier_in is None:
+            module.fail_json(msg="If you specify failover, region or weight you must also specify identifier")
+        if (weight_in is None and region_in is None and failover_in is None) and identifier_in is not None:
+            module.fail_json(msg="You have specified identifier which makes sense only if you specify one of: weight, region or failover.")
 
     if vpc_id_in and not private_zone_in:
-        module.fail_json(msg="parameter 'private_zone' must be true when specifying parameter"
-            " 'vpc_id'")
-
+        module.fail_json(msg="parameter 'private_zone' must be true when specifying parameter 'vpc_id'")
```

Comment:
I agree...I'm not quite sure why it was originally written this way but I can just have it default private_zone to True if vpc_id is not None.